kiwi/runtime_checker.py

```diff
@@ -412,6 +414,23 @@ def check_efi_mode_for_disk_overlay_correctly_setup(self):
         if overlayroot and firmware == 'uefi':
             raise KiwiRuntimeError(message)
 
+    def check_grub_efi_installed_for_efi_firmware(self):
+        """
+        If the image is being built with efi or uefi firmware setting
+        we need the grub2-x86_64-efi package installed
+        """
+        message = dedent('''\n
+            Firmware set to efi or uefi but the grub2-x86_64-efi is not
+            part of the image build.
+        ''')
+        firmware = self.xml_state.build_type.get_firmware()
+        if firmware in ('efi', 'uefi'):
+            package_names = \
+                self.xml_state.get_bootstrap_packages() + \
+                self.xml_state.get_system_packages()
+            if 'grub2-x86_64-efi' not in package_names:
```

Comment:
This looks all good and works now. I just tried to avoid checking for explicit package names in the runtime checks because there is no standard for this name across the distributions. Therefore this most probably will fail for e.g centos or other distros and there is no way to switch off the runtime check.

I suggest you run this check only if the configured package manager is set to zypper, which kind of means this is a suse build and you can match for the package safer